perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -483,6 +527,74 @@ public void setAttributes(PerunSession sess, Resource resource, Member member, L
 		setAttributes(sess, resource, member, attributes, false);
 	}
 
+	@Override
+	public void setAttributes(PerunSession sess, Member member, Group group, List<Attribute> attributes) throws InternalErrorException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException, UserNotExistsException {
+		setAttributes(sess, member, group, attributes, false);
+	}
+
+	@Override
+	public void setAttributes(PerunSession sess, Member member, Group group, List<Attribute> attributes, boolean workWithUserAttributes) throws InternalErrorException, WrongAttributeValueException, WrongAttributeAssignmentException, WrongReferenceAttributeValueException, UserNotExistsException {
+		// classification of attributes to attributes to remove and attributes to set
+		List<Attribute> attributesToRemove = new ArrayList<Attribute>();
+		List<Attribute> attributesToSet = new ArrayList<Attribute>();
+		for(Attribute attribute : attributes) {
+			if (attribute.getValue() == null) {
+				attributesToRemove.add(attribute);
+			} else {
+				attributesToSet.add(attribute);
+			}
+		}
+		removeAttributes(sess, member, group, attributesToRemove, workWithUserAttributes);
+		// fist we have to store attributes into DB because checkAttributesValue can be preformed only on stored attributes.
+		if (!workWithUserAttributes) {
+			long timer = Utils.startTimer();
+			for (Attribute attribute : attributesToSet) {
+				//skip core attributes
+				if (!getAttributesManagerImpl().isCoreAttribute(sess, attribute)) {
+					setAttributeWithoutCheck(sess, member, group, attribute, false);
+				}
+			}
+			log.debug("addMember timer: setAttributes (for(Attribute attribute : attributes)) [{}].", Utils.getRunningTime(timer));
+		} else {
+			long timer = Utils.startTimer();
+			User user = getPerunBl().getUsersManagerBl().getUserByMember(sess, member);
+			log.debug("addMember timer: getFacility and User [{}].", Utils.getRunningTime(timer));
+
+			for(Attribute attribute : attributesToSet) {
+				boolean changed = false;
+				// skip core attributes
+				if(!getAttributesManagerImpl().isCoreAttribute(sess, attribute)) {
+
+					if(getAttributesManagerImpl().isFromNamespace(sess, attribute, AttributesManager.NS_MEMBER_GROUP_ATTR)) {
+						timer = Utils.startTimer();
+						changed = setAttributeWithoutCheck(sess, member, group, attribute, false);
```

Comment:
I am not sure there, but if this method really has boolean attribute (workWithUserAttribute), maybe there is no need for this condition, cause this method may be able to do it by itself. Please check it.